plugins/trayicon.py

```diff
@@ -129,21 +129,22 @@ def window_state_event_cb(self, widget, event):
         "Hide window when minimize"
         if event.changed_mask & event.new_window_state & Gdk.WindowState.ICONIFIED:
             self.window.deiconify()
+            self.shell.save_position()
             self.window.hide()
 
     def trayicon_click(self, widget, data = None):
-        self.window.deiconify()
-        self.window.show()
+        self.shell.toggle_visibility()
```

Comment:
This change will not work. Actually there should be a large comment there about why (my fault for it being missing)...

The rationale is that forcing showing Liferea on any tray click is the only way to support all the hidden window bug reports we've seen over the last 2 years. There were users who could not access the application anymore because there desktop environment caused the visibility check to fail and the app never showed up again.

While it would be great to have toggling for all the people where it works properly, it cannot be for all those broken setups...

Reply:
Got it. I remember this used to be the behavior and had a vague recollection that there was some reason it wasn't this way anymore, but couldn't remember why. I'll leave a comment.

I didn't want to duplicate the logic in here but part of the reasoning for this change is to make sure that we restore the saved window position (if any). That logic is private at the moment but I'll work something out.

Reply:
Thanks!